reaper/manager.go

```diff
@@ -29,70 +92,112 @@ type TaskReaper struct {
 	// DB
 	DB *gorm.DB
 	// k8s client.
-	Client client.Client
+	Client k8s.Client
 }
 
 //
 // Run Executes the reaper.
 // Rules by state:
 //   Created
-//   - Deleted after the defined period.
+//   - Deleted after TTL.Created > created timestamp or
+//     settings.Task.Reaper.Created.
+//   Pending
+//   - Deleted after TTL.Pending > created timestamp or
+//     settings.Task.Reaper.Created.
+//   Postponed
+//   - Deleted after TTL.Postponed > created timestamp or
+//     settings.Task.Reaper.Created.
+//   Running
+//   - Deleted after TTL.Running > started timestamp.
 //   Succeeded
-//   - Deleted after the defined period when not associated with
-//     an application. Tasks associated with an application
-//     are deleted when the application is deleted.
+//   - Deleted after TTL > terminated timestamp or
+//     settings.Task.Reaper.Succeeded.
 //   - Bucket is released after the defined period.
 //   - Pod is deleted after the defined period.
 //   Failed
-//   - Deleted after the defined period when not associated with
-//     an application. Tasks associated with an application
-//     are deleted when the application is deleted.
+//   - Deleted after TTL > terminated timestamp or
+//     settings.Task.Reaper.Failed.
 //   - Bucket is released after the defined period.
 //   - Pod is deleted after the defined period.
 func (r *TaskReaper) Run() {
```

Comment:
Add support for TTL.
The reaper strategy changed to only delete tasks when a TTL is defined explicitly.  Else, the reaper will only _release_ the task resources.

Note: For analysis tasks, the TTL should never be defined for the Succeeded state.